### PR TITLE
fix: add support for Node.js v18.6.0

### LINKED ISSUE #36: `ERR_LOADER_CHAIN_INCOMPLETE` on Node.js 18.6
After updating to Node.js 18.6.0 I start to have an error:

```
> tsm node_modules/uvu/bin.js . '\.test\.(ts|js)$'

(node:111446) ExperimentalWarning: Custom ESM Loaders is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:111446) DeprecationWarning: Obsolete loader hook(s) supplied and will be ignored: getFormat, transformSource
Error [ERR_LOADER_CHAIN_INCOMPLETE]: "file:///home/ai/Dev/nanostores/node_modules/.pnpm/tsm@2.2.1/node_modules/tsm/loader.mjs 'load'" did not call the next hook in its chain and did not explicitly signal a short circuit. If this is intentional, include `shortCircuit: true` in the hook's return.
    at new NodeError (node:internal/errors:387:5)
    at ESMLoader.load (node:internal/modules/esm/loader:624:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async ESMLoader.moduleProvider (node:internal/modules/esm/loader:472:11)
    at async link (node:internal/modules/esm/module_job:67:21)
```

It looks like a conflict with a new loader API from [Node.js 18.6](https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V18.md#18.6.0)

### PR DESCRIPTION
**Pull request by antongolub**
closes #36

https://github.com/microsoft/playwright/pull/15694/files

### CODE REVIEW

**Comment by antongolub** [comment]
btw, `esbuld >= 0.14.32` breaks tsc tests

File: package.json [hunk 1]
```diff
@@ -30,7 +30,8 @@
   },
   "scripts": {
     "build": "node build",
-    "types": "tsc"
+    "types": "tsc",
+    "test": "node bin.js test/config/index.ts --tsmconfig test/config/tsm.js && node bin.js test/index.mjs && node -r ./require.js test/config/index.ts --tsmconfig test/config/tsm.js && node -r ./require.js test/index.js && node --loader ./loader.mjs test/config/index.ts --tsmconfig test/config/tsm.js && node --loader ./loader.mjs test/index.mjs"
```
Comment (lukeed) [inline package.json:34]:
```suggestion
    "types": "tsc"
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ jobs:
     timeout-minutes: 3
     strategy:
       matrix:
-        nodejs: [12, 14, 16.11, 16]
+        nodejs: [12, 14, 16.11, 16, 18.6]
```
Comment (lukeed) [inline .github/workflows/ci.yml:18]:
```suggestion
        nodejs: [12, 14, 16.11, 16, 18]
```

File: src/loader.ts [hunk 3]
```diff
@@ -111,9 +119,9 @@ export const resolve: Resolve = async function (ident, context, fallback) {
 	}
 
 	return fallback(ident, context, fallback);
-}
+})
 
-export const load: Load = async function (uri, context, fallback) {
+export const load: Load = withShortCircuit(async function (uri, context, fallback) {
```
Comment (lukeed) [inline src/loader.ts:124]: Can you remove the wrapper function & just add `shortCircuit: true` to the `return`ed object literals; eg https://github.com/lukeed/tsm/blob/44f9108f95b8554348fb6602d07aea47081edf0b/src/loader.ts#L92

```suggestion
export const load: Load = async function (uri, context, fallback) {
```

**Review by lukeed — CHANGES_REQUESTED** [review]

**Comment by lukeed** [comment]
Why they didn't add this flag with a `true` default value is beyond me. So annoying... intra-version breaking change. Node team continues to ace the ESM rollout 🙄

**Comment by antongolub** [comment]
ESM is our python3-like frustrative chanllenge)

@lukeed, Do you know what is returnd by `fallback`.
Should we use `return Object.assign(await fallback(...), {shortCircuit: true})` or `return fallback(...)` woulb be enought?

File: src/loader.ts (hunk 3 above)
Comment (antongolub) [reply src/loader.ts]: Done

**Review by lukeed — APPROVED** [review]
Calling `fallback` invokes/passes off to native behavior, so whatever the normal hook would have done will/should be done. But at this point who knows anymore

**Pull request merged**